pkg/driverbuilder/templates.go

```diff
@@ -90,7 +93,101 @@ func LoadMakefileObjList(c builder.Config) (string, error) {
 			return strings.Split(l, "@PROBE_NAME@-y += ")[1], nil
 		}
 	}
-	return "", fmt.Errorf("obj list not found")
+	return "", fmt.Errorf("kmod obj list not found")
+}
+
+type makefileBpfData struct {
+	MakeObjList string
+}
+
+const makefileBpfTemplate = `
+always-y += probe.o
+always = $(always-y)
+
+LLC ?= llc
+CLANG ?= clang
+
+KERNELDIR ?= /lib/modules/$(shell uname -r)/build
+
+NEEDS_COS_73_WORKAROUND = $(shell expr ` + "`" + `grep -sc "^\s*struct\s\+audit_task_info\s\+\*audit;\s*$$" $(KERNELDIR)/include/linux/sched.h` + "`" + ` = 1) 
+ifeq ($(NEEDS_COS_73_WORKAROUND), 1)
+	KBUILD_CPPFLAGS += -DCOS_73_WORKAROUND
+endif
+
+IS_CLANG_OLDER_THAN_10 := $(shell expr ` + "`$(CLANG) -dumpversion | cut -f1 -d.`" + ` \<= 10)
+ifeq ($(IS_CLANG_OLDER_THAN_10), 1)
+	KBUILD_CPPFLAGS := $(filter-out -fmacro-prefix-map=%,$(KBUILD_CPPFLAGS))
+endif
+
+all:
+	make -C $(KERNELDIR) M=$$PWD
+
+clean:
+	make -C $(KERNELDIR) M=$$PWD clean
+	@rm -f *~
+
+$(obj)/probe.o: {{ .MakeObjList }}
+	$(CLANG) $(LINUXINCLUDE) \
+		$(KBUILD_CPPFLAGS) \
+		$(KBUILD_EXTRA_CPPFLAGS) \
+		$(DEBUG) \
+		-I.. \
+		-D__KERNEL__ \
+		-D__BPF_TRACING__ \
+		-Wno-gnu-variable-sized-type-not-at-end \
+		-Wno-address-of-packed-member \
+		-fno-jump-tables \
+		-fno-stack-protector \
+		-Wno-tautological-compare \
+		-O2 -g -emit-llvm -c $< -o $(patsubst %.o,%.ll,$@)
+	$(LLC) -march=bpf -filetype=obj -o $@ $(patsubst %.o,%.ll,$@)
+`
+
+func renderBpfMakefile(w io.Writer, md makefileBpfData) error {
+	t := template.New("bpf-makefile")
+	t, _ = t.Parse(makefileBpfTemplate)
+	return t.Execute(w, md)
+}
+
+func LoadBpfMakefile(buffer *bytes.Buffer, c builder.Config) error {
+	bpfMakefile, err := downloadFile(fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/driver/bpf/Makefile", c.RepoOrg, c.RepoName, c.DriverVersion))
+	if err == nil {
+		// Existent bpf/Makefile! (old libs versions, pre https://github.com/falcosecurity/libs/pull/1188)
+		_, err = buffer.WriteString(bpfMakefile)
+		return err
+	}
+	objList, err := loadBpfMakefileObjList(c)
+	if err != nil {
+		return err
+	}
+	return renderBpfMakefile(buffer, makefileBpfData{MakeObjList: objList})
+}
+
+func loadBpfMakefileObjList(c builder.Config) (string, error) {
+	parsedCmakeLists, err := downloadFile(fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/driver/bpf/CMakeLists.txt", c.RepoOrg, c.RepoName, c.DriverVersion))
+	if err != nil {
+		return "", err
+	}
+	lines := strings.Split(parsedCmakeLists, "\n")
+	startDeps := false
+	var deps string
+	for _, l := range lines {
+		l = strings.TrimSpace(l)
+		if strings.HasPrefix(l, "set(BPF_SOURCES") {
```

Comment:
Yep this code is really ugly :( it's so hard to deal with new and old versions, without breaking backward compatibility.
Btw for now this PR is on hold because it is only needed if https://github.com/falcosecurity/libs/pull/1188 PR gets merged in libs :)